server/lib/notifications.js

```diff
@@ -172,7 +231,7 @@ async function notifyByEmail(activity) {
       break;
 
     case activityType.COLLECTIVE_EXPENSE_CREATED:
-      notifyAdminsOfCollective(activity.data.collective.id, activity);
+      notifyAdminsOfCollectiveAndCheckUserExpensesAmount(activity);
```

Comment:
this looks like overloading a function. I'd rather see two separate functions.
`notifyAdminsOfCollective` can be performed independently of the action of the bot.

Reply:
Humm Ok, I wrapped them up because i thought the order was important, first sending an email to the `admin` the creating the `comment` to trigger the `new comment` email. Isn't it?

Reply:
Keeping functions small and decoupling them is generally a good idea.
If something fails, it doesn't fail everything. Also easier to maintain.

The order here doesn't matter. Basically, whenever a new comment is created another service should pick up this event and do a bunch of things. We could even imagine different micro services, one for notifying followers of that thread (admins), one for each bot.
Even though we don't have those micro services them today, it's good to have that in mind and build our code in a way that will be easier to migrate to such architecture in the future.